spec/preloading_spec.cr

```diff
@@ -0,0 +1,90 @@
+require "./spec_helper"
+
+describe "Preloading" do
+  it "can disable lazy loading" do
+    with_lazy_load(enabled: false) do
+      post = PostBox.save
+
+      posts = Post::BaseQuery.new
+
+      expect_raises LuckyRecord::LazyLoadError do
+        posts.first.comments
+      end
+    end
+  end
+
+  it "preloads has_many" do
+    with_lazy_load(enabled: false) do
+      post = PostBox.save
+      comment = CommentBox.new.post_id(post.id).save!
+
+      posts = Post::BaseQuery.new.preload_comments
+
+      posts.results.first.comments.should eq([comment])
+    end
+  end
+
+  it "preloads has_many with custom query" do
+    with_lazy_load(enabled: false) do
+      post = PostBox.save
+      comment = CommentBox.new.post_id(post.id).save!
+
+      posts = Post::BaseQuery.new.preload_comments(
+        Comment::BaseQuery.new.id.not(comment.id)
+      )
+
+      posts.results.first.comments.should eq([] of Comment)
+    end
+  end
+
+  it "works with nested preloads" do
+    with_lazy_load(enabled: false) do
+      post = PostBox.save
+      comment = CommentBox.new.post_id(post.id).save!
+
+      posts = Post::BaseQuery.new.preload_comments(
+        Comment::BaseQuery.new.preload_post
+      )
+
+      posts.first.comments.first.post.should eq(post)
+    end
+  end
+
+  it "uses an empty array if there are no associated records" do
+    with_lazy_load(enabled: false) do
+      post = PostBox.save
+
+      posts = Post::BaseQuery.new.preload_comments
+
+      posts.results.first.comments.should eq([] of Comment)
+    end
+  end
+
+  it "preloads belongs_to" do
+    with_lazy_load(enabled: false) do
+      post = PostBox.save
+      comment = CommentBox.new.post_id(post.id).save!
+
+      comments = Comment::BaseQuery.new.preload_post
+
+      comments.first.post.should eq(post)
+    end
+  end
+
+  it "preloads optional belongs_to" do
+  end
+end
+
+private def with_lazy_load(enabled)
+  begin
+    LuckyRecord::Model.configure do
+      settings.lazy_load_enabled = enabled
```

Comment:
Yeah I think it would be cool to add something to the luckyframework/habitat that allows you to set configuration that's rolled back at the end. Maybe something like this?

```crystal
# Or maybe temp_config?
LuckyRecord::Model.test_config lazy_load_enabled: false do
  # do stuff
end
```